ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/management-api-gnmi.j2

```diff
@@ -36,6 +36,38 @@ management api gnmi
 {%                 endif %}
 {%             endfor %}
 {%         endif %}
+{%         if management_api_gnmi.transport.grpc_tunnel is arista.avd.defined %}
+{%             for transport in management_api_gnmi.transport.grpc_tunnel %}
+{%                 if transport.name is arista.avd.defined %}
+   transport grpc-tunnel {{ transport.name }}
+{%                 endif %}
```

Comment:
By making the `name` key the primary key - it is now required and so you don't need this `if` statement anymore. Note that if someone was omitting the name here you could still be rendering shutdown and so on..

```suggestion
   transport grpc-tunnel {{ transport.name }}
```

I know this is done (almost) like this above but this is because since schema introduction some templates would need to be cleaned up (though if you take a closer look to the example above you can notice that the `if` around name is actually one level higher than the if statements for the other settings).

Reply:
Looks like this one still needs to be addressed!